src/navigation/reroute.js

```diff
@@ -71,6 +72,25 @@ export function reroute(pendingPromises = [], eventArguments) {
     });
   }
 
+  /**
+   * Let's imagine that some kind of delay occurred during application loading.
+   * The user without waiting for the application to load switched to another route,
+   * this means that we shouldn't bootstrap and mount that application, thus we check
+   * twice if that application should be active before bootstrapping and mounting.
+   * https://github.com/single-spa/single-spa/issues/524
+   */
+  function tryToBootstrapAndMount(app, unmountAllPromise) {
```

Comment:
Since this function doesn't use any variables inside of the outer function scope, let's move it outside of the reroute function. Maybe the best place for it is just at the bottom of this file, as a top level function within the file. That way it won't be recreated every time reroute is called, and also makes it a bit clear to maintainers that it's a function that doesn't rely on anything except its arguments.

Reply:
Okay, no problem.